src/protocol/http/org/apache/jmeter/protocol/http/curl/BasicCurlParser.java

```diff
@@ -226,19 +189,23 @@ public void setCompressed(boolean compressed) {
             this.compressed = compressed;
         }
 
+        /**
+         * @param name the field of Header
+         * @param value the value of Header
+         */
         public void addHeader(String name, String value) {
-            if (name.equalsIgnoreCase("COOKIE")) {
-               this.cookieInHeaders=value;
-            }
-            else if (!NOT_ADD_HEADERS.contains(name)) {
+            if (name != null && name.equalsIgnoreCase("COOKIE")) {
```

Comment:
Actually, my intention to place `"COOKIE"` in front, was to make the `null` check unnecessary here.
`if ("COOKIE".equalsIgnoreCase(name)) { ... }` would be enough.
If a `null` value is problematic for the set-contains check, we could return early from this method when `name` is `null`.